### PR TITLE
[fix](optimizer) Fix sql block when new optimizer is enabled

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

The check would skipped since when checkBlockPolicy get invoked, new optimizer doesn't do plan yet

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.01 seconds
 stream load tsv:          531 seconds loaded 74807831229 Bytes, about 134 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17162213470 Bytes

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 49.12 seconds
 stream load tsv:          528 seconds loaded 74807831229 Bytes, about 135 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17161925956 Bytes

**Comment by Kikyou1997** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.28 seconds
 stream load tsv:          535 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162308824 Bytes

File: regression-test/suites/sql_block_rule_p0/test_sql_block_rule.groovy [hunk 1]
```diff
@@ -47,6 +47,15 @@ suite("test_sql_block_rule") {
...
     }
+    test {
+        sql("EXPLAIN SELECT * FROM table_2", false)
+        exception "sql match regex sql block rule: test_rule_sql"
+    }
+
+    test {
+        sql("INSERT INTO table_2 SELECT * FROM table_2", false)
+        exception "sql match regex sql block rule: test_rule_sql"
+    }
```
Comment (morrySnow) [inline regression-test/suites/sql_block_rule_p0/test_sql_block_rule.groovy:58]: add check partition num, tablet num, cardinality cases

**Comment by Kikyou1997** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**